torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -106,13 +106,29 @@ template<typename... Args> inline variable_list flatten_tensor_args(Args&&... ar
 
 // See NOTE [ Autograd View Variables ] for details.
 inline Tensor as_view(const Tensor & base, Tensor tensor, bool is_differentiable,
-                      CreationMeta creation_meta=CreationMeta::DEFAULT) {
+        c10::optional<std::function<Tensor(const Tensor&)>> func=c10::nullopt,
+        CreationMeta creation_meta=CreationMeta::DEFAULT) {
   auto base_var = Variable(base);
   if (base_var.is_view()) {
+    // Set `func` using the root base as input.
+    // `func` is used to recover views in backward when as_strided is not supported.
+    // See Note [View + Inplace update on base tensor] and [View + Inplace update on view tensor]
+    // for more details how we use this function in backward.
+    if (func.has_value()) {
+      auto fn = func.value();
+      auto diff_view_meta = static_cast<DifferentiableViewMeta*>(torch::autograd::impl::get_autograd_meta(base_var));
+      if (!diff_view_meta->support_as_strided()) {
```

Comment:
Not sure about this check.
Do we support views that are mixed between types that support strided and don't?
- If no then this should be an assert
- If yes, then the else should create a prev_fn using as_strided and do the same trick.

I would lean towards no here as we don't have any plan to mix these right now !

Reply:
@albanD So I put a `TORCH_CHECK` in the lambda function since we 
- We cannot assert outside lambda, it's totally valid to view on `unbind` outputs. 
- We should only error out when inplace update happens 
So now this function is being called when we refresh `grad_fn()` in `variable.cpp`, we throw an error for it.  
Not this behavior is not consistent with current PT behaviorhttps://github.com/pytorch/pytorch/issues/36608. I made XLA behavior to be the desired behavior we want. It's being tested here https://github.com/pytorch/xla/pull/1891/files#diff-76c04ebee87386cd778382d7198e9800R1008
Since changing PT behavior is a BC-breaking change, I'd leave it out of scope for this PR for now. 
Let me know if this is fine for you. Thanks!

Reply:
[Ehhh I accidentally deleted my reply while I wanted to edit it :( ]
@albanD I put a `TORCH_CHECK` inside lambda function since 
- We cannot assert outside lambda function as it's totally valid to view on `unbind` outputs.
- We should only error out when inplace update happens.

Currently this function errors out when we refresh `grad_fn()` in `variable.cpp` when inplace update happens. So this is indeed the desired behavior. It's being tested in https://github.com/pytorch/xla/pull/1891/files#diff-76c04ebee87386cd778382d7198e9800R1008
Note it's actually not consistent with current PT behavior due to issue https://github.com/pytorch/pytorch/issues/36608. Since changing PT behavior is BC-breaking change, I'd leave it out of scope for this PR for now. 
Please let me know if this is fine for you. Thanks!

Reply:
Ok the new code looks good to me. And using `has_view_fn()` makes it less confusing !